cmd/nvidia_gpu/nvidia_gpu.go

```diff
@@ -41,22 +43,52 @@ var (
 	enableContainerGPUMetrics      = flag.Bool("enable-container-gpu-metrics", false, "If true, the device plugin will expose GPU metrics for containers with allocated GPU")
 	gpuMetricsPort                 = flag.Int("gpu-metrics-port", 2112, "POrt on which GPU metrics for containers are exposed")
 	gpuMetricsCollectionIntervalMs = flag.Int("gpu-metrics-collection-interval", 30000, "Colection interval (in milli seconds) for container GPU metrics")
+	gpuConfigFile                  = flag.String("gpu-config", "/etc/nvidia/gpu_config.json", "File with GPU configurations for device plugin")
 )
 
+func parseGPUConfig(gpuConfigFile string) (gpumanager.GPUConfig, error) {
```

Comment:
Yup, but we still need this. Since partitioning the GPU will be done separately, we need a way to make the device plugin wait until all the partitions are created, before registering the resources with the kubelet. So we need the gpu config to tell the device plugin what the desired state of partitions should be.

Reply:
uh got it, so this config is actually a result of configuring GPU partitions.